### PR TITLE
[CH]Update clickhouse backend doc to 23.1

### PR DESCRIPTION
**Pull request by loneylee**

### CODE REVIEW

**Comment by zhztheplayer** [comment]
The CI failure during step `Setup docker container` might not be related. A fix proposed at #1036

File: docs/ClickHouse.md [hunk 1]
```diff
@@ -33,30 +33,25 @@ In general, we use IDEA for Gluten development and CLion for ClickHouse backend
...
-    sudo apt install -y clang-12 lldb-12 lld-12 clang-12-doc llvm-12-doc llvm-12-examples clang-tools-12 libclang-12-dev clang-format-12 libfuzzer-12-dev libc++-12-dev libc++abi-12-dev libllvm-12-ocaml-dev
-
-    sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 100 --slave /usr/bin/clang++ clang++ /usr/bin/clang++-12
-
-    sudo update-alternatives --config clang  # choose the clang-12
-    clang --version  # check the version of the clang
-```
+    ```
+
+- Clang 15.0 or higher version ( Please refer to [How-to-Build-ClickHouse-on-Linux](https://clickhouse.com/docs/en/development/build/) )
```
Comment (zzcclp) [inline docs/ClickHouse.md:38]: modify: Clang 14.0

**Review by zzcclp — APPROVED** [review]
LGTM

File: docs/ClickHouse.md (hunk 1 above)
Comment (zzcclp) [reply docs/ClickHouse.md:38]: Got it.

**Pull request merged**